test/integration/auth.spec.ts

```diff
@@ -428,6 +434,148 @@ describe('admin.auth', () => {
+    before(function() {
+      /* tslint:disable:no-console */
+      if (!cmdArgs.testMultiTenancy) {
+        // To enable, run: npm run test:integration -- --testMultiTenancy
+        // By default we skip multi-tenancy as it is a Google Cloud Identity Platform
+        // feature only and requires to be enabled via the Cloud Console.
+        console.log(chalk.yellow('    Skipping multi-tenancy tests.'));
+        this.skip();
+      }
+      /* tslint:enable:no-console */
+    });
+
+    // Delete test tenants at the end of test suite.
+    after(() => {
+      const promises: Array<Promise<any>> = [];
+      createdTenants.forEach((tenantId) => {
+        promises.push(
+            admin.auth().deleteTenant(tenantId).catch((error) => {/** Ignore. */}));
+      });
+      return Promise.all(promises);
+    });
+
+    it('createTenant() should resolve with a new tenant', () => {
+      return admin.auth().createTenant(tenantOptions)
+        .then((actualTenant) => {
+          createdTenantId = actualTenant.tenantId;
+          createdTenants.push(createdTenantId);
+          expectedCreatedTenant.tenantId = createdTenantId;
+          expect(actualTenant.toJSON()).to.deep.equal(expectedCreatedTenant);
+        });
+    });
+
+    it('getTenant() should resolve with expected tenant', () => {
+      return admin.auth().getTenant(createdTenantId)
+        .then((actualTenant) => {
+          expect(actualTenant.toJSON()).to.deep.equal(expectedCreatedTenant);
+        });
+    });
+
+    it('updateTenant() should resolve with the updated tenant', () => {
+      expectedUpdatedTenant.tenantId = createdTenantId;
+      expectedUpdatedTenant2.tenantId = createdTenantId;
+      const updatedOptions: admin.auth.UpdateTenantRequest = {
+        displayName: expectedUpdatedTenant.displayName,
+        emailSignInConfig: {
+          enabled: false,
+        },
+      };
+      const updatedOptions2: admin.auth.UpdateTenantRequest = {
+        emailSignInConfig: {
+          enabled: true,
+          passwordRequired: false,
+        },
+      };
+      return admin.auth().updateTenant(createdTenantId, updatedOptions)
+        .then((actualTenant) => {
+          expect(actualTenant.toJSON()).to.deep.equal(expectedUpdatedTenant);
+          return admin.auth().updateTenant(createdTenantId, updatedOptions2);
+        })
+        .then((actualTenant) => {
+          expect(actualTenant.toJSON()).to.deep.equal(expectedUpdatedTenant2);
+        });
+    });
+
+    it('listTenants() should resolve with expected number of tenants', () => {
+      const tenantOptions2 = deepCopy(tenantOptions);
+      tenantOptions2.displayName = 'testTenant2';
+      return admin.auth().createTenant(tenantOptions2)
+        .then((actualTenant) => {
+          createdTenants.push(actualTenant.tenantId);
+          // Test listTenant pagination.
+          return admin.auth().listTenants(1);
+        })
+        .then((result) => {
```

Comment:
Something we do in other SDKs to test list operations is to list everything, and see if the known results appear in the output. Something like:

```
const allTenantIds = listAndIterateThroughAllTenants();
// scan allTenantIds to see if createdTenants is a subset of it.
```

Reply:
Done